### PR TITLE
Replace underpowered git-user-name by native implementation and fix error handling

### PR DESCRIPTION
**Pull request by micgro42**
The package [`git-user-name`] merely parsed the global git configuration without evaluating any includes therein or other configuration (system, local).

This new implementation uses the git cli directly. The git cli can be assumed available since it is also used by gitlog.

Further, this implements actually working error handling. If there is an error, e.g. no git username is found or the git binary is not in `$PATH`, then the error message is shown in the daily and weekly commits sections. 
This resolves the bug that the implementation didn't throw an error if there was no username. Also, even if an error was thrown by the library, the error message was not printed where the user could see it, but the application merely exited (which is mentioned as an anti-pattern in nodejs' [process documentation]).

[`git-user-name`]: https://github.com/jonschlinkert/git-user-name/blob/22cf1d3880356c754f7d84dec242987a3b8fcc7e/index.js
[process documentation]: https://nodejs.org/api/process.html#process_process_exit_code

### CODE REVIEW

File: gitbot.js [hunk 1]
```diff
@@ -59,9 +69,15 @@ function findGitRepos(repos, depth, callback) {
...
  */
-function getCommitsFromRepos(repos, days, callback) {
+async function getCommitsFromRepos(repos, days, callback) {
   let cmts = [];
-  let gitUser = getGitUser();
+  let gitUser;
+  try {
+    gitUser = await getGitUser();
+  } catch( error ) {
+    callback(error);
```
Comment (mojoaxel) [inline gitbot.js]: If the error is returned here we propably need error-handling in the function that calls `getCommitsFromRepos`!?

File: gitbot.js [hunk 2]
```diff
@@ -3,18 +3,28 @@ const subdirs = require('subdirs');
...
-  }
+      \n${error.message}\n${stdout}\n`;
+}
+
+async function getGitUser() {
+  return new Promise((resolvePromise, rejectPromise) => {
+    exec('git config --get user.name', (error, stdout) => {
+
+      if (error || stdout.trim() === '') {
+        rejectPromise(craftGetGitUserErrorMessage(error, stdout));
```
Comment (mojoaxel) [inline gitbot.js]: Because the function `craftGetGitUserErrorMessage` is used no where else I would recommend embedding its code here directly.

**Review by mojoaxel — COMMENTED** [review]
> The package git-user-name merely parsed the global git configuration without evaluating any includes therein or other configuration (system, local).

:+1: You should open a bug-report there!

File: gitbot.js (hunk 1 above)
Comment (mojoaxel) [reply gitbot.js]: I see...so the error is just displayed. I think this is a valid solution! Thx!

**Review by mojoaxel — APPROVED** [review]
@micgro42 Thank you! Very nice! :tada:

File: gitbot.js (hunk 2 above)
Comment (micgro42) [reply gitbot.js]: My motivation for splitting out that function was indeed not reusability, but readability. That way, both `getGitUser` and `craftGetGitUserErrorMessage` only have a single responsibility.

**Pull request merged**